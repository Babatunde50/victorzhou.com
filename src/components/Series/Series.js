// @flow
import React from 'react';
import SeriesPost from './SeriesPost';
import Author from '../Author';
import Discuss from '../Discuss';
import Share from '../Share';
import SubscribeForm from '../SubscribeForm';

type Props = {|
  +frontmatter: Object,
  +html: string,
  +htmlEnd: string,
  +seriesPosts: {
    +edges: Array<Object>,
  },
|};

const Series = ({ frontmatter, html, htmlEnd, seriesPosts }: Props) => {
  const {
    discussLinkTwitter,
    discussLinkHN,
    discussLinkReddit,
    isML,
    isWeb,
    seriesSlugs,
    slug,
    title,
  } = frontmatter;

  // seriesPosts aren't necessarily in order, so we sort by seriesSlugs
  const frontmatters = seriesPosts.edges.map(e => e.node.frontmatter);
  frontmatters.sort((a, b) => seriesSlugs.indexOf(a.slug) - seriesSlugs.indexOf(b.slug));

  return (
    <div>
      <div dangerouslySetInnerHTML={{ __html: html }} />
      {frontmatters.map((post, i) => (
        <SeriesPost key={post.title} {...post} n={i + 1} />
      ))}
      <div dangerouslySetInnerHTML={{ __html: htmlEnd }} />
      <SubscribeForm signupSource={`Series:${slug}`} isML={isML} isWeb={isWeb} large />
      <Author showBio showTwitter />
      <Share url={slug} title={title} shareText="SHARE THIS SERIES" />
      <Discuss twitter={discussLinkTwitter} hn={discussLinkHN} reddit={discussLinkReddit} />
    </div>
  );
};

export default Series;
