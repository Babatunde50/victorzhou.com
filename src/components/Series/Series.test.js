import React from 'react';
import renderer from 'react-test-renderer';
import Series from './Series';

describe('Series', () => {
  const props = {
    htmlEnd: '<div></div>',
    series: {
      frontmatter: {
        date: new Date(99999),
        slug: 'test',
        title: 'test',
      },
      html: '<div></div>',
    },
    seriesPosts: {
      edges: [{
        node: {
          frontmatter: {
            title: 'test',
            date: new Date(99999),
          },
        },
      }],
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Series {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
