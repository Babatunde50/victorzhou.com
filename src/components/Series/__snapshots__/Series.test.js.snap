// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series renders correctly 1`] = `
<div
  className="post"
>
  <article
    className="content"
  >
    <h1
      className="content__title no-subtitle"
    >
      test
    </h1>
    <div
      className="content__date"
    >
      <p
        className="content-date"
      >
        <time>
          December 31, 1969
        </time>
      </p>
    </div>
    <div
      className="content__body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div></div>",
        }
      }
    />
    <div>
      <div
        className="series-post"
      >
        <div
          className="series-post-content"
        >
          <a>
            <img />
          </a>
          <div>
            <h3>
              <a>
                1. test
              </a>
            </h3>
            <p
              className="content-date"
            >
              <time>
                December 31, 1969
              </time>
            </p>
            <p />
          </div>
        </div>
      </div>
      <div
        className="content__body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
    </div>
  </article>
  <div
    className="post__subscribeForm"
  >
    <div
      className="container large "
    >
      <p
        className="description"
      >
        I write about ML, Web Dev, and more. 
        <b>
          Subscribe to get new posts by email!
        </b>
      </p>
      <form
        acceptCharset="utf-8"
        action="https://sendy.victorzhou.com/subscribe"
        method="post"
        target="_blank"
      >
        <input
          name="Source"
          type="hidden"
          value="Post:test"
        />
        <input
          name="list"
          type="hidden"
          value="CWC7638hEb6mfk1RqUbJ763snA"
        />
        <input
          autoCapitalize="off"
          autoCorrect="off"
          name="hp"
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex="-1"
          type="text"
        />
        <input
          aria-label="Email Address"
          autoCapitalize="off"
          autoCorrect="off"
          name="email"
          placeholder="example@domain.com"
          size="25"
          type="email"
        />
        <br />
        <input
          type="submit"
          value="SUBMIT"
        />
      </form>
    </div>
  </div>
  <div
    className="post__viewSource"
  >
    <p>
      <i>
        This blog is
         
        <a
          href="https://github.com/vzhou842/victorzhou.com"
          target="_blank"
        >
          open-source on Github
        </a>
        .
      </i>
    </p>
  </div>
  <div
    className="post__footer"
  >
    <div
      className="container-small"
    />
    <div
      className="post__authorContainer"
    >
      <StaticQuery
        render={[Function]}
      />
    </div>
    <StaticQuery
      render={[Function]}
    />
  </div>
  <div
    className="post__comments"
  >
    <div
      style={
        Object {
          "paddingBottom": "320px",
        }
      }
    />
  </div>
</div>
`;
